### DISCUSSION TITLE
[solved] Arch Linux & Kotlin Support

### DISCUSSION THREAD

**Discussion by AIGLEZMA** [wiverson/maven-jpackage-template#30] [Q&A] [answered]
I have tried to use this template with kotlin and tornadofx, but i keep getting this error when executing the install command

`Error: Invalid or unsupported type: [null]`

my pom.xml: https://gist.github.com/AigleDev/be997184c3155a31554b9349fd571752

**Comment by wiverson** [comment, marked as answer]
Not sure. I don't use Kotlin or TornadoFX. If you post more of your error log I might be able to help / point you in the right direction.

**Reply by AIGLEZMA** [reply]
https://gist.github.com/AigleDev/f070d8283eebe6471a8c416a6a7acb38

**Reply by wiverson** [reply]
Looks like jpackage is looking for a type argument.

https://docs.oracle.com/en/java/javase/14/docs/specs/man/jpackage.html

Try setting a valid package type for your OS manually in linux-jpackage.txt

**Reply by AIGLEZMA** [reply]
I'm on Arch Linux and tried all of them ("rpm", "deb", "pkg") and none of them work

**Reply by wiverson** [reply]
Huh. Not sure, could try asking on a TornadoFX forum. I don't do Arch Linux, Kotlin, or TornadoFX so you're getting pretty far afield from what I can help you with...

**Reply by AIGLEZMA** [reply]
Alright got it working, first for arch linux the type is "app-image" and tornadofx + kotlin works perfectly.

Thank you!

**Reply by wiverson** [reply]
Fantastic!  :)

**Reply by abulka** [reply]
@AIGLEZMA Any tips on integrating Kotlin into this template? Presumably some Kotlin jars would need to be added and the Maven recipe updated somehow?

**Reply by AIGLEZMA** [reply]
> @AIGLEZMA Any tips on integrating Kotlin into this template? Presumably some Kotlin jars would need to be added and the Maven recipe updated somehow?

You just have to add kotlin stdlib to dependencies and the kotlin compiler plugin
example here: https://gist.github.com/AIGLEZMA/6b9f19345fc7e6811daa67da1a642024
